tests/unit/s2n_client_cert_verify_test.c

```diff
@@ -42,6 +47,66 @@ static int test_sign(const struct s2n_pkey *priv_key, s2n_signature_algorithm si
     return S2N_SUCCESS;
 }
 
+void *pkey_task(void *param)
+{
+    struct s2n_cert_chain_and_key *chain_and_key = s2n_connection_get_selected_cert(pkey_conn);
+    EXPECT_NOT_NULL(chain_and_key);
+
+    s2n_cert_private_key *pkey = s2n_cert_chain_and_key_get_private_key(chain_and_key);
+    EXPECT_NOT_NULL(pkey);
+
+    EXPECT_SUCCESS(s2n_async_pkey_op_perform(pkey_op, pkey));
+    pthread_exit(NULL);
+}
+
+int async_pkey_perform_op(struct s2n_connection *conn, struct s2n_async_pkey_op *op)
+{
+    EXPECT_NOT_NULL(conn);
+    EXPECT_NOT_NULL(op);
+
+    pkey_op = op;
+    pkey_conn = conn;
+
+    POSIX_GUARD(pthread_create(&worker, NULL, &pkey_task, NULL));
+
+    return S2N_SUCCESS;
+}
+
+static int negotiate(struct s2n_connection *conn, s2n_blocked_status *block) 
```

Comment:
To avoid confusion / help with debugging stack traces, can you name your helper methods more descriptively? Like negotiate -> s2n_test_negotiate_with_async_pkey_op and pkey_task -> s2n_test_pkey_task

Reply:
Have gone and updated the names to be more descriptive